docs/feature_midi.md

```diff
@@ -0,0 +1,299 @@
+| MI_CH10                    | 5CA8      | Go to channel 10                                        |
+| MI_CH11                    | 5CA9      | Go to channel 11                                        |
+| MI_CH12                    | 5CAA      | Go to channel 12                                        |
+| MI_CH13                    | 5CAB      | Go to channel 13                                        |
+| MI_CH14                    | 5CAC      | Go to channel 14                                        |
+| MI_CH15                    | 5CAD      | Go to channel 15                                        |
+| MI_CH16                    | 5CAE      | Go to channel 16                                        |
+| MI_CHD                     | 5CAF      | Go to channel 17                                        |
+| MI_CHU                     | 5CB0      | Go to channel 18                                        |
+| MI_ALLOFF                  | 5CB1      | Turn off all notes                                      |
+| MI_SUS                     | 5CB2      | Sustain                                                 |
+| MI_PORT                    | 5CB3      | Portmento                                               |
+| MI_SOST                    | 5CB4      | Sostenuto                                               |
+| MI_SOFT                    | 5CB5      | Soft Pedal                                              |
+| MI_LEG                     | 5CB6      | Legato                                                  |
+| MI_MOD                     | 5CB7      | Modulation                                              |
+| MI_MODSD                   | 5CB8      | Decrease Modulation Speed                               |
+| MI_MODSU                   | 5CB9      | Increase Modulation Speed                               |
+| MI_BENDD                   | 5CBA      | Bend pitch down                                         |
+| MI_BENDU                   | 5CBB      | Bend pitch up                                           |
+
+### Background
+
+QMK MIDI is based on TMK MIDI which in turn is based on `avr-midi`
+
+QMK MIDI is a subset implementation of the MIDI 1.0 specification. 
+
+### References
+
+#### QMK Internals 
+ 
+ * [Internals/MIDI Device Setup Process](internals_midi_device_setup_process.md)
+ * [Internals/MIDI Device](internals_midi_device.md)
+ * [Internals/MIDI Util](internals_midi_util.md)
+ 
+ 
+#### QMK C Files
+
+ * `quantum/process_keycode/process_midi.c`
+ * `quantum/quantum_keycodes.h`
+ * `tmk_core/protocol/midi.h`
+ * `tmk_core/protocol/midi.c`
+ * `tmk_core/protocol/qmk_midi.c`
+ * `tmk_core/protocol/midi_device.h`
+
+#### MIDI Specification
+
+ * [MIDI.org](https://www.midi.org/specifications-old/item/table-1-summary-of-midi-message)
+ * [CMU MIDI Programmer's Reference](https://www.cs.cmu.edu/~music/cmsip/readings/MIDI%20tutorial%20for%20programmers.html)
```

Comment:
```suggestion
```

Reply:
Why remove the reference section? The documentation isn't complete yet, only covering what's commonly used in keymaps. I think it's useful to have the reference, especially since CC codes aren't defined anywhere in QMK, only in the MIDI specification, and they're the second most commonly used way to interact with the MIDI specification outside of the keycodes.

Reply:
The background section is not terribly useful information, and the internals and C files sections are also subject to change (the internals stuff is technically autogenerated from doc comments but it's half baked).

Reply:
I think removing the background section is fine, but the c files haven't changed in ~4 years, and it's useful to know where they are.

Reply:
Again, the autogenerated docs should really be doing that job. It just requires the code to be documented in the correct way for Doxygen to pick it up, and possibly some refactoring.